server/src/Types.hs

```diff
@@ -86,10 +89,25 @@ newEnvIO serverOptions =
     >>= Aeson.eitherDecodeFileStrict @Shelley.ProtocolParameters
     <&> second (Env serverOptions)
 
+getNodeConnectInfo :: AppM (C.LocalNodeConnectInfo C.CardanoMode)
+getNodeConnectInfo =
+  asks serverOptions <&> \opts ->
+    C.LocalNodeConnectInfo
+      { localConsensusModeParams =
+          -- FIXME: Calc Byron epoch length based on Genesis params.
+          C.CardanoModeParams (C.EpochSlots 21600)
+      , localNodeNetworkId = networkId opts
+      , localNodeSocketPath = nodeSocket opts
+      }
+
 newtype Cbor = Cbor Text
   deriving stock (Show)
   deriving newtype (Eq, FromHttpApiData, ToHttpApiData, FromJSON, ToJSON)
 
+newtype ExecutionUnitsMap = ExecutionUnitsMap ([(Cbor, Cbor)])
```

Comment:
Since the ctor is exported it's possible to construct invalid values for this type. Wouldn't defining it as `ExecutionUnitsMap [(TxWitness.RdmrPtr, Scripts.ExUnits)]` be worthwhile?
Also, you don't need the parens here.